packages/geolocator/tizen/src/permission_manager.cc

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "permission_manager.h"
+
+#include <Ecore.h>
+#include <privacy_privilege_manager.h>
+
+#include <vector>
+
+#include "log.h"
+
+namespace {
+
+constexpr char kPrivilegeLocation[] = "http://tizen.org/privilege/location";
+constexpr char kPrivilegeLocationCoarse[] =
+    "http://tizen.org/privilege/location.coarse";
+
+struct PermissionResponse {
+  ppm_call_cause_e cause;
+  ppm_request_result_e result;
+  bool received = false;
+};
+
+}  // namespace
+
+PermissionManager::PermissionManager() {}
+PermissionManager::~PermissionManager() {}
+
+TizenResult PermissionManager::CheckPermissionStatus(
+    PermissionStatus *permission_status) {
+  ppm_check_result_e check_result;
+
+  int result = ppm_check_permission(kPrivilegeLocation, &check_result);
+  if (result != PRIVACY_PRIVILEGE_MANAGER_ERROR_NONE) {
+    return TizenResult(result);
+  }
+
+  switch (check_result) {
+    case PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_DENY:
+    case PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_ASK:
+      *permission_status = PermissionStatus::kDenied;
+      break;
+    case PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_ALLOW:
+    default:
+      *permission_status = PermissionStatus::kAlways;
+      break;
+  }
+  return TizenResult();
+}
+
+void PermissionManager::RequestPermssion(const OnSuccess &on_success,
+                                         const OnFailure &on_failure) {
+  std::vector<std::string> permissions = {kPrivilegeLocation,
+                                          kPrivilegeLocationCoarse};
+  std::vector<PermissionStatus> results;
+
+  int ret = 0;
+  for (size_t i = 0; i < permissions.size(); i++) {
```

Comment:
Do you need to request multiple permissions?

According to [the documentation](https://docs.tizen.org/application/native/tutorials/details/sec-privileges), the both privileges share the same permission group (Location).

https://docs.tizen.org/application/native/guides/security/privacy-related-permissions/

> Since the privileges are grouped, the user’s decision regarding one privilege applies to the whole group of related privileges. For example, if the user has granted permission to use the `http://tizen.org/privilege/account.read` privilege, permission is automatically granted to the `http://tizen.org/privilege/account.write` privilege also. Be aware that both privileges need to be declared in the application manifest file. If you declare only one of them, the above rule does not apply.

Reply:
No, you don't have to, but I requested all, even so, only one permission pop-up actually appears.